pandas/tests/groupby/test_groupby.py

```diff
@@ -1954,3 +1954,39 @@ def test_groupby_only_none_group():
     expected = pd.Series([np.nan], name="x")
 
     assert_series_equal(actual, expected)
+
+
+@pytest.mark.parametrize("bool_agg_func", ["any", "all"])
+def test_bool_aggs_dup_column_labels(bool_agg_func):
+    # 21668
+    df = pd.DataFrame([[True, True]], columns=["a", "a"])
+    grp_by = df.groupby([0])
+    result = getattr(grp_by, bool_agg_func)()
+
+    expected = df
+    tm.assert_frame_equal(result, expected)
+
+
+@pytest.mark.parametrize(
+    "idx", [pd.Index(["a", "a"]), pd.MultiIndex.from_tuples((("a", "a"), ("a", "a")))]
+)
+def test_dup_labels_output_shape(groupby_func, idx):
```

Comment:
which of these fail in master?  the first couple i tried passed

Reply:
For reasons I'm not sure of some of these stopped failing on master but started giving incorrect output. any/all are good examples of that. Any transformation function (bfill, ffill, etc... - you can find these all in base.py) also gives incorrect output. `nunique` continues to fail.

Reply:
> stopped failing on master but started giving incorrect output.

Not sure I follow.  Does that mean that these tests are not testing for correct output?

Reply:
The tests are correct. On master these only give one column output, even though there are two columns, save nunique which continues to fail.